splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/SortOperationIT.java

```diff
@@ -263,6 +264,27 @@ public void testSortOperationByKey2Descending() throws Exception {
 		Assert.assertEquals("Incorrect name ordering!",reversedByValue,returnedByValue);
 	}
 
+	@Test(expected = SQLSyntaxErrorException.class)
+	public void testDerby6027() throws Exception {
+		methodWatcher.executeQuery("values 1,2 order by int(1)");
+	}
+
+	@Test(expected = SQLSyntaxErrorException.class)
+	public void orderByWithComputation() throws Exception {
+		methodWatcher.executeQuery("values 1 order by 1+0");
```

Comment:
You could catch the exception here and assert its sql code is the new one you added.